app/models/like.rb

```diff
@@ -19,6 +19,14 @@ def relayable_options
 
   after_destroy do
     self.parent.update_likes_counter
+    participation = Participation.where(target_id: target_id, author_id: author_id).first
+    if participation.present?
+      if participation.count == 1
+        participation.destroy
+      else
+        participation.update!(count: participation.count.pred)
+      end
+    end
```

Comment:
You're still duplicating code here, I'd like to see that extracted somewhere, possibly a class method on `Participation`.